scripts/linters/pylint_extensions.py

```diff
@@ -611,6 +627,66 @@ def visit_functiondef(self, node):
         self.check_functiondef_yields(node, node_doc)
         self.check_docstring_style(node)
         self.check_docstring_section_indentation(node)
+        self.check_typeinfo(node, node_doc)
+
+    def check_typeinfo(self, node, node_doc):
+        """Checks whether all parameters in a function definition are
+        properly formatted.
+
+        Args:
+            node: astroid.node.Function. Node for a function or
+                method definition in the AST.
+            node_doc: Docstring. Pylint Docstring class instance representing
+                a node's docstring.
+        """
+        re_param_line = re.compile(
```

Comment:
This regex looks very hard to interpret -- I'm not able to guess much from this.

Reply:
I've taken this regex from the `docstring_checks` file and modify it, which in turn is taken from the pylint codebase. This is the basic structure we need to check docstring parameters.

Reply:
Can you please add a comment on top of both of these regex linking to the pylint codebase from where it was taken?

Reply:
Done.